test/core/util/memory_counters.cc

```diff
@@ -42,40 +42,42 @@ static void guard_free(void* vptr);
 #endif
 
 static void* guard_malloc(size_t size) {
-  size_t* ptr;
+  uintptr_t* ptr;
   if (!size) return nullptr;
   NO_BARRIER_FETCH_ADD(&g_memory_counters.total_size_absolute, (gpr_atm)size);
   NO_BARRIER_FETCH_ADD(&g_memory_counters.total_size_relative, (gpr_atm)size);
   NO_BARRIER_FETCH_ADD(&g_memory_counters.total_allocs_absolute, (gpr_atm)1);
   NO_BARRIER_FETCH_ADD(&g_memory_counters.total_allocs_relative, (gpr_atm)1);
-  ptr = static_cast<size_t*>(g_old_allocs.malloc_fn(size + sizeof(size)));
-  *ptr++ = size;
-  return ptr;
+  ptr = static_cast<uintptr_t*>(
+      g_old_allocs.malloc_fn(size + sizeof(uintptr_t) * 2));
```

Comment:
Instead of this magic formula, can we add a common macro somewhere called something like `ROUND_UP_TO_ALIGNMENT_SIZE()` and use it here?

Reply:
Sure, but that's what the macro will have :-) But that's fine with me.